src/engines/TrainDieselElectricEngine.cpp

```diff
@@ -3,7 +3,7 @@
 namespace godot {
     void TrainDieselElectricEngine::_bind_methods() {}
 
-    TEngineType TrainDieselElectricEngine::get_engine_type() {
-        return TEngineType::DieselElectric;
+    TrainEngine::EngineType TrainDieselElectricEngine::get_engine_type() {
+        return TrainEngine::EngineType::ENGINE_TYPE_DIESEL_ELECTRIC;
```

Comment:
Consider `TrainEngine::Type::TYPE_DIESEL_ELECTRIC` or even shorter `TrainEngine::Type::DIESEL_ELECTRIC` (please compare to TrainDoors.hpp)

AFAIK enums are scoped in classes, and we're already in the engine context (TrainEngine). Also engine types are quite unique, so they shouldn't clash with other enums in this scope.